batterycelltesting/OCV.py

```diff
@@ -30,50 +27,47 @@ def Graphing (filename):
    ax.yaxis.set_major_formatter(mtick.FormatStrFormatter('%.6f'))
 
    plt.scatter(current,voltage)
-   plt.savefig('OCVDataGraph.png', bbox_inches='tight') ##Creating a new file each
+   plt.savefig((filename) + '.png', bbox_inches='tight') ##Creating a new file each
    plt.clf()
    
-def Calculation():
+def calculation():
    stdV = np.std(voltage)
    avgV = np.mean(voltage)
    print("Average Voltages for all samples: %s" % avgV)
-   ValidVoltage = []
-   ValidCurrent = []
+   valid_voltage = []
+   valid_current = []
    Weights = []
    i = 0
    for v in voltage:
-       if v < avgV+3*stdV and v > avgV-3*stdV:
-           ValidVoltage.append(voltage[i]) ##Filtering outlier Voltages out
-           ValidCurrent.append(current[i])
+       if v < avgV + 3 * stdV and v > avgV - 3 * stdV:
+           valid_voltage.append(voltage[i]) ##Filtering outlier Voltages out
+           valid_current.append(current[i])
        i = i + 1
 
 
-   for x in ValidCurrent:
-       Weights.append(Weighting(x,ValidCurrent)) #Adds a weighting based on current
+   for x in valid_current:
+       Weights.append(weighting(x,valid_current)) #Adds a weighting based on current
        
    weightedsumV = 0
-   for i in range(0, len(ValidVoltage)):
-      weightedsumV = weightedsumV + ValidVoltage[i]*Weights[i]
+   for i in range(0, len(valid_voltage)):
+      weightedsumV = weightedsumV + valid_voltage[i] * Weights[i]
       
-   weightavgV = [weightedsumV/sum(Weights)] #Divide by the sum of the weights,
-   #not amount of elements, big issue I had.
+   weightavgV = [weightedsumV / sum(Weights)] 
    return weightavgV
    print("Weighted Average for Filtered Voltages: %s"  % weightavgV)
    
-#print(Valids[0][0]) #First number is index, second is Voltage Or Current or Weight
-
 
 
 outputs = []
 for filename in os.listdir():
-    if filenameendswith.("OCV.csv"):
+    if filename.endswith("OCVData.csv"):
```

Comment:
Main issue here is that the file only works with the OCVData, we'd want it to work with any file of that type. Try instead to use argparser